firefox_ui_harness/runtests.py

```diff
@@ -4,18 +4,37 @@
 
 import copy
 import sys
+import tempfile
 
-from marionette import BaseMarionetteTestRunner
-from marionette.runtests import cli
+import mozinstall
 
+from marionette import (BaseMarionetteOptions,
+                        BaseMarionetteTestRunner,)
+from mozlog import structured
+
+import firefox_puppeteer
 import firefox_ui_tests
 
-from .arguments import ReleaseTestParser
 from .default_prefs import default_prefs
 from .testcase import FirefoxTestCase
 
 
-class ReleaseTestRunner(BaseMarionetteTestRunner):
+class FirefoxUIOptions(BaseMarionetteOptions):
+
+    def parse_args(self, *args, **kwargs):
+        options, test_files = BaseMarionetteOptions.parse_args(self,
+                                                               *args, **kwargs)
+
+        if not any([(k.startswith('log_') and v is not None and '-' in v)
+                    for (k, v) in vars(options).items()]):
+            options.log_mach = '-'
```

Comment:
This if can be skipped if you specify {"mach": sys.stdout} instead of "tbpl" below.